### PR TITLE
CIV-8254 Unit Tests coverage and Refactor

### PR DESCRIPTION
**Pull request by UshaPanneerselvam1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-8254

### Change description ###
Unit Test coverage for the God classes

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java [hunk 1]
```diff
@@ -688,6 +692,7 @@ private void assembleResponseDocuments(CaseData caseData, CaseData.CaseDataBuild
         }
     }
 
+    //This line will not be reached when AddApplicant2 Is No, so can never be fully covered
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java:695]: I'll need to understand how you got to this conclusion.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandlerTest.java [hunk 2]
```diff
@@ -483,7 +479,169 @@ void shouldValidateExperts_whenMultipartyAndSolicitorRepresentsOnlyOneOfResponde
 
             assertThat(response.getErrors()).isEmpty();
         }
+        @Test
+        void shouldValidateExperts_whenMultipartyAndSolicitorRepresentsOnlyOneOfRespondents() {
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandlerTest.java:483]: How are you checking that the experts are being validated?

According to the assertion at the end, you're just checking that there are no errors but that doesn't guarantee that any validation has been carried out.

Let's check what part of code you're trying to validate and rework this test together.

**Review by filippo-possenti-version1 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java (hunk 1 above)
Comment (UshaPanneerselvam1) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java]: Have added a Positive scenario

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandlerTest.java (hunk 2 above)
Comment (UshaPanneerselvam1) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandlerTest.java:509]: Done

**Review by filippo-possenti-version1 — APPROVED** [review]

**Review by drummondjm — APPROVED** [review]

**Pull request merged**